fc/parsing/actions.py

```diff
@@ -780,6 +864,99 @@ def sympy_equations(self):
                 eqs.append(eq.to_sympy(self._symbol_generator, self._number_generator))
         return self._sympy_equations
 
+    #######################################
+    # Helper methods for model manipulation
+
+    def _convert_time_if_needed(self):
+        """Units-convert the time variable if not in the protocol's units."""
+        time_units = self.units.get_unit(self.time_units)
+        time_var = self.model.get_free_variable_symbol()
+        time_var = self.model.convert_variable(time_var, time_units, DataDirectionFlow.INPUT)
+
+    def _add_input_variables(self):
+        """Ensure input variables exist in the desired units.
+
+        TODO: At present this just checks they exist; no units conversion or initial_value handling.
+        TODO: If an input doesn't exist but has units & initial_value here, or is defined via
+        an equation, then add a new variable to the model.
+        """
+        for var in self.inputs:
+            assert var.units is None
```

Comment:
Why no units?

Reply:
Because the case where there are units specified isn't implemented yet.